packages/@pollyjs/core/src/utils/normalize-request.js

```diff
@@ -27,7 +28,11 @@ export function url(url, config = {}) {
     parsedUrl.set('query', sortedQuery);
   }
 
-  return parsedUrl.href;
+  if (isRelative) {
```

Comment:
Lets put this behind a config option. We can enable it by default but at least users will have the ability to turn this functionality off. 

```js
{ 
  matchRequestsBy: { 
    url: { 
      relative: true 
    } 
  } 
}
```

_config name not final_